From ff0e62f83efbf080175e9cd8055befe649fe09f5 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Mon, 9 Jul 2018 14:06:00 +0800
Subject: [PATCH] arm64: support pack zboot.img

zboot.img: Image.lz4 + resouorce.img
boot.img:  Image + resource.img

Change-Id: I97bc39de99ce85cc21f2491a6db59d1e015f67d0
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm64/Makefile | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 53cde97203a2..7f9be09fba1d 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -163,7 +163,7 @@ define archhelp
   echo  '                  install to $$(INSTALL_PATH) and run lilo'
 endef
 
-kernel.img: Image
+kernel.img: Image.lz4
 	$(Q)$(srctree)/scripts/mkkrnlimg $(objtree)/arch/arm64/boot/Image $(objtree)/kernel.img >/dev/null
 	@echo '  Image:  kernel.img is ready'
 ifdef CONFIG_MODULES
@@ -177,6 +177,17 @@ LOGO_KERNEL := $(notdir $(wildcard $(srctree)/logo_kernel.bmp))
 	$(Q) if [ -f $(srctree)/$(LOGO_KERNEL) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO_KERNEL) $(objtree)/; fi
 	$(Q)$(srctree)/scripts/resource_tool $(objtree)/arch/arm64/boot/dts/rockchip/$*.dtb $(LOGO) $(LOGO_KERNEL)
 	@echo '  Image:  resource.img (with $*.dtb $(LOGO) $(LOGO_KERNEL)) is ready'
-	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then $(srctree)/scripts/mkbootimg --kernel $(objtree)/arch/arm64/boot/Image --second resource.img -o boot.img && echo '  Image:  boot.img (with Image resource.img) is ready'; fi
-
-CLEAN_FILES += boot.img kernel.img resource.img
+	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then \
+		$(srctree)/scripts/mkbootimg \
+			--kernel $(objtree)/arch/arm64/boot/Image \
+			--second resource.img \
+			-o boot.img && \
+		echo '  Image:  boot.img (with Image resource.img) is ready'; \
+		$(srctree)/scripts/mkbootimg \
+			--kernel $(objtree)/arch/arm64/boot/Image.lz4 \
+			--second resource.img \
+			-o zboot.img && \
+		echo '  Image:  zboot.img (with Image.lz4 resource.img) is ready'; \
+	fi
+
+CLEAN_FILES += boot.img kernel.img resource.img zboot.img
-- 
2.35.3

